src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -539,6 +539,13 @@ private void RemoveLocalScopes(LocalSymbol local)
             return null;
         }
 
+        public override BoundNode? VisitCompoundAssignmentOperator(BoundCompoundAssignmentOperator node)
+        {
+            base.VisitCompoundAssignmentOperator(node);
+            ValidateAssignment(node.Syntax, node.Left, node, isRef: false, _diagnostics);
```

Comment:
It seems reasonable enough to have the whole compound-assignment node stand in for the `node.Left op node.Right` (plus conversions) that we would ideally want to use.